local/src/core/http2.cc

```diff
@@ -697,11 +697,48 @@ receive_nghttp2_request(
 }
 
 static int
-on_frame_send_cb(
-    nghttp2_session * /* session */,
-    nghttp2_frame const * /* frame */,
-    void * /* user_data */)
+on_frame_send_cb(nghttp2_session *session, nghttp2_frame const *frame, void *user_data)
 {
+  auto *session_data = reinterpret_cast<H2Session *>(user_data);
+  auto const stream_id = frame->hd.stream_id;
+  auto stream_map_iter = session_data->_stream_map.find(stream_id);
+  if (stream_map_iter == session_data->_stream_map.end()) {
+    // Nothing to do if this is not in our stream map.
+    return 0;
+  }
+  H2StreamState &stream_state = *stream_map_iter->second;
+  Errata errata;
+
+  errata.note(
+      S_DIAG,
+      "Sent frame for key {}: {}",
+      stream_state._key,
+      H2FrameNames[static_cast<H2Frame>(frame->hd.type)]);
+
+  if (stream_id != 0) {
+    auto const frame_type = stream_state._client_abort_frame == -1 ?
+                                stream_state._server_abort_frame :
+                                stream_state._client_abort_frame;
+    auto const error_code = stream_state._client_abort_error == -1 ?
+                                stream_state._server_abort_error :
+                                stream_state._client_abort_error;
+    if (frame->hd.type == frame_type) {
+      errata.note(
+          S_DIAG,
+          "Submitting RST_STREAM frame for key {} with error code {}.",
```

Comment:
Now that you helped me understand that `frame_type` is the previous frame, let's put this in the log here as well. Something like:

```
Submitting RST_STREAM frame for key {} after {} frame with error code {}."